### PR TITLE
Force convert dimension values while loading from the url state

### PR DESCRIPTION
**Pull request by AdityaHegde**
We are facing issues with data type mismatch from the url state object and the actual dimension type. This was fixed on main but older urls with broken state are not handled.

This PR force converts the dimension values from the url state to keep things consistent.

### CODE REVIEW

File: runtime/queries/metricsview_data_types.go
Comment (begelundmuller) [inline runtime/queries/metricsview_data_types.go:18]: In the alerts PR, I'm making some refactors to stop passing full metrics view specs in here. Would recommend doing the same here (can wait until the other PR is merged so the correct funcs are available in this package). See [this diff](https://github.com/rilldata/rill/pull/3783/files#diff-4a01247b922ef6106d44f5553dc3fba7130fadb020de3ff8260e29b71f9c4e8e) for `runtime/queries/metricsview_aggregation.go` for an example.

File: runtime/queries/metricsview_data_types.go
Comment (begelundmuller) [inline runtime/queries/metricsview_data_types.go:27]: Key prefix needs updating

File: runtime/queries/metricsview_data_types.go
Comment (begelundmuller) [inline runtime/queries/metricsview_data_types.go:35]: The deps should only return immediate dependencies. The responsibility to resolve the DAG should be an upstream concern (at the moment, this is handled by invalidating the cache based on `StateUpdatedOn`, which is increased in all cases where a ref changes).

File: proto/rill/runtime/v1/queries.proto
Comment (begelundmuller) [inline proto/rill/runtime/v1/queries.proto:534]: It would be preferable to return a `rill.runtime.v1.StructType`, which is our generic type for representing schemas (e.g. see `MetricsViewAggregationResponse`). Then it will work predictably for both DuckDB and Druid (and any other OLAP backends we add in future).

File: runtime/queries/metricsview_data_types.go
Comment (begelundmuller) [inline runtime/queries/metricsview_data_types.go:77]: As addressed previously, would suggest two refactors:

1. Return a `rill.runtime.v1.StructType` instead of raw type names, so we have predictable types for any OLAP backend (so the frontend doesn't get tied too closely to DuckDB type names specifically). It's available in the `Schema` property of the `drivers.Result` type returned from OLAP queries – it should be possible to construct a query that doesn't scan any data, but returns all dims and measures as fields just to get the types.
2. Make sure it also supports Druid – we need to support it as a first-class citizen for all metrics-related queries

File: proto/rill/runtime/v1/queries.proto
Comment (begelundmuller) [inline proto/rill/runtime/v1/queries.proto:99]: Maybe just call the API `MetricsViewTypes` or `MetricsViewSchema`? One less word ¯\_(ツ)_/¯

Comment (AdityaHegde) [reply proto/rill/runtime/v1/queries.proto:99]: `MetricsViewSchema` Sounds better

File: runtime/queries/metricsview_schema.go
Comment (begelundmuller) [inline runtime/queries/metricsview_schema.go:116]: 1. Use `safeName(q.TableName)` to escape it correctly
2. I wonder if putting the `LIMIT 1` at the end will cause a full table scan? Maybe doing `SELECT ... FROM (SELECT * FROM tbl LIMIT 1) GROUP BY ...` might protect against that? (Also curious if `LIMIT 0` would be adequate enough to still get the schema back?)

Comment (AdityaHegde) [reply runtime/queries/metricsview_schema.go]: Updated to use `LIMIT 0`, was the fasted based on brenchmarks.

**Comment by ericpgreen2** [comment]
Leaving a note that we discussed:

With the new call to `MetricsViewSchema`, it's not ideal that we need (another) network request before loading a dashboard. In the future, we could at least batch the call with the `TimeRangeSummary` request, but we should consider if these could be avoided outright.

**Comment by ericpgreen2** [comment]
Approving from the frontend perspective, so we can test on the staging environment with our list of old URLs.

**Review by ericpgreen2 — APPROVED** [review]

**Pull request merged**
